### PR TITLE
Claude Code: Seed .claude instructions for working on the Python extension kernel

### PR DESCRIPTION
**Pull request by wesm**
Without these, when you ask Claude to help with the Python extension (and the positron module in particular), it has to do some work to discover the right approach to run the unit tests etc., and it can go down dark alleys. These instructions can surely be made much better

### CODE REVIEW

File: .claude/positron-python.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+## Project Structure
+```
+extensions/positron-python/
+├── src/                           # TypeScript extension code
+├── python_files/                  # Python runtime components
+│   ├── posit/                    # Positron-specific Python modules
+│   │   ├── positron/             # Core Positron Python functionality
+│   │   │   └── tests/            # Python unit tests for Positron modules
+│   │   └── pyproject.toml        # Python project configuration
+│   └── lib/                      # Vendored Python dependencies
```
Comment (seeM) [inline .claude/positron-python.md:16]: We may want to include the `positron` test requirements file as well. I totally forgot which file is recommended - @isabelizimm do you remember?

File: .claude/positron-python.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+### Test Requirements
+- **Module Resolution**: Always run pytest from `python_files/posit` directory
+- **No Main Blocks**: Never use `if __name__ == "__main__"` in test files
+
+## Code Quality & Linting
+
+### Automatic Formatting
+Code formatting is handled automatically via Claude Code hooks using ruff format.
```
Comment (seeM) [inline .claude/positron-python.md:91]: Interesting, how do I configure these hooks?

File: .claude/positron-python.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+# Type checking with Pyright (from python_files/)
+pyright
+```
+
+### Tool Versions & Configuration
+- **Pyright**: Version 1.1.308 (matches CI)
+- **Ruff**: Line length 100 characters, target Python 3.8+
+- **Enabled rules**: Comprehensive set including flake8-bugbear, pycodestyle, isort, etc.
+- **Excludes**: Vendored dependencies (`lib/`, `posit/positron/_vendor/`)
```
Comment (seeM) [inline .claude/positron-python.md]: Have you found these necessary? I would have assumed that CC would only need to know how to run the tools and that we could hide these details

File: .claude/positron-python.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+## Architecture & Key Components
+
+### Python Runtime Integration
+- **Kernel Management**: IPython kernel lifecycle and communication
+- **Environment Detection**: Virtual environments, conda, pyenv, pipenv, poetry
+- **Data Science Features**: Interactive data exploration and visualization
+- **Language Server**: Pylsp/Jedi integration for IntelliSense
+- **Debugging**: Integration with debugpy for Python debugging
+
```
Comment (seeM) [inline .claude/positron-python.md]: I feel like we could refine this but I'm not sure how specifically this section is trying to steer CC

File: .claude/positron-python.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        ([1, 2, 3, 4, 5], "expected_result_1"),
+        ([1.1, 2.2, 3.3], "expected_result_2"),
+        # Edge cases
+        ([], "empty_result"),
+        ([1], "single_item_result"),
+    ]
+    
+    for input_data, expected in test_cases:
+        result = function_under_test(input_data)
+        assert result == expected
```
Comment (seeM) [inline .claude/positron-python.md]: We haven't discussed our team preference, but my habit thus far has been to use pytest parameters, since it gives more control over running and debugging specific test cases. Type annotations aren't strictly needed but can catch type errors in the tests and could further guide CC:

```suggestion
@pytest.mark.parametrize(
    ("input_data", "expected"),
    [
        # Regular cases
        ([1, 2, 3, 4, 5], "expected_result_1"),
        ([1.1, 2.2, 3.3], "expected_result_2"),
        # Edge cases
        ([], "empty_result"),
        ([1], "single_item_result"),
    ],
)
def test_function_with_various_inputs(input_data: List[float], expected: str):
    """Test function with different input types and edge cases."""
    result = function_under_test(input_data)
    assert result == expected
```

**Review by seeM — DISMISSED** [review]
Thanks @wesm for taking the initiative and setting up our Claude team infrastructure! I'm sure many of us have been experimenting and glad we now have a platform to share that.

**For future:** How do you handle permissions for bash commands like running tests? Should we consider shared settings files with well-scoped commands allowed?

Should we consider helper scripts e.g. one that runs ruff check, format check, and pyright? Perhaps another script that runs ruff check --fix and format? Etc.

File: .claude/positron-python.md (hunk 2 above)
Comment (wesm) [reply .claude/positron-python.md:91]: It's in the .claude/hooks directory, pretty clever actually. https://github.com/posit-dev/positron/commit/a547aa33d3f69f9d79fd2bd5188b0275d66135fd

File: .claude/positron-python.md (hunk 4 above)
Comment (wesm) [reply .claude/positron-python.md]: I'll just remove this generated section since it doesn't add much value

File: .claude/positron-python.md (hunk 3 above)
Comment (wesm) [reply .claude/positron-python.md]: Yeah, I'm just going to delete this unnecessary detail

File: .claude/positron-python.md (hunk 1 above)
Comment (wesm) [reply .claude/positron-python.md:16]: Will add

**Comment by wesm** [comment]
> Should we consider helper scripts e.g. one that runs ruff check, format check, and pyright? Perhaps another script that runs ruff check --fix and format? Etc.

I took a pass on this and added some helper scripts. Let me know if what I added is what you had in mind?

**Review by seeM — DISMISSED** [review]
Looks good, thanks! The outline of helper scripts is what I had in mind. We can add the implementations separately if needed.

**Comment by wesm** [comment]
I guess I forgot to check the scripts in! I’ll rebase and add those and then merge
